src/ProjectTemplates/Shared/Project.cs

```diff
@@ -466,7 +466,7 @@ private void CaptureBinLogOnFailure(ProcessEx result)
         if (result.ExitCode != 0 && !string.IsNullOrEmpty(ArtifactsLogDir))
         {
             var sourceFile = Path.Combine(TemplateOutputDir, "msbuild.binlog");
-            Assert.True(File.Exists(sourceFile), $"Log for '{ProjectName}' not found in '{sourceFile}'.");
+            Assert.True(File.Exists(sourceFile), $"Log for '{ProjectName}' not found in '{sourceFile}'. Execution output:{result.Output}~Error:~{result.Error}~{result}");
```

Comment:
I'm tempted to keep this part of the logs long term, I feel like it could assist with debugging.

For context; the current output with this is:

```
[1.501s] [Templates.Blazor.Test.EmptyBlazorServerTemplateTest] [Error] Test threw an exception.
Xunit.Sdk.TrueException: Log for 'AspNet.Rqm5pu3hoaxg' not found in '/datadisks/disk1/work/A35D08E8/w/A83309A0/e/Templates/BaseFolder/AspNet.Rqm5pu3hoaxg/msbuild.binlog'. Execution output:Microsoft (R) Build Engine version 17.3.0-preview-22256-03+72a076794 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.
~Error:~~Microsoft.AspNetCore.Internal.ProcessEx
```

We can probably change this to:

```suggestion
            Assert.True(File.Exists(sourceFile), $"Log for '{ProjectName}' not found in '{sourceFile}'. Execution output: {result.Output}");
```